app/models/manageiq/providers/autosde/storage_manager/volume_mapping.rb

```diff
@@ -0,0 +1,14 @@
+class ManageIQ::Providers::Autosde::StorageManager::VolumeMapping < ::VolumeMapping
```

Comment:
@gregoryb-at-ibm I just noticed that the volume_mappings table doesn't have a `:type` column so STI isn't possible.

Can you put in a PR to add that column to volume_mappings?

Reply:
Ah! Must have missed it...
How come it still works and I can create volume-mapping?
And how come we don't have `:type` in `host_initiators`?

Here's the schema PR https://github.com/ManageIQ/manageiq-schema/pull/558

Reply:
> How come it still works and I can create volume-mapping?

Probably because you're using class_by_ems which is returning a valid class and calling a method off of it, but that class isn't actually backed by the database record.

E.g.:
```
host_initiator = HostInitiator.class_by_ems(ems)
host_initiator.raw_create_volume # "works"
HostInitiator.find(:id).raw_create_volume # would blow up
```

If you have a subclass for `HostInitiator` then it also needs a type